app/build.gradle

```diff
@@ -1,19 +1,55 @@
+project.ext.set('usesGms', false)
+project.ext.set('usesHms', false)
+
 apply plugin: 'com.android.application'
 
 android {
     compileSdkVersion 29
     defaultConfig {
         applicationId "com.dimelo.sampleapp"
-        minSdkVersion 16
+        minSdkVersion 17
         targetSdkVersion 29
         versionCode 6
         versionName "1.1"
     }
+
+    /*signingConfigs {
+        config {
+            storeFile file('xxx.jks')
+            keyAlias 'xxx'
+            keyPassword 'xxx'
+            storePassword 'xxx'
+            v1SigningEnabled true
+            v2SigningEnabled true
+        }
+    }*/
+
+    flavorDimensions "vendor"
+    productFlavors {
+        google {
+            dimension "vendor"
+            versionNameSuffix "-google"
+//            apply plugin: "com.google.gms.google-services"
```

Comment:
Does this mean that we need to manually comment / uncomment the correct "apply plugin" before generating the APK? 

Shouldn't we have the automatic detection on our sample app so that we can generate the same APK for any device? This is similar to building the multiple APK for QA on release, the goal is to reduce this work.

Reply:
it is already an automatic detection but I commented this line because I had a compilation problem with **FCM** since we decided with driss to remove the `googlo-service.json` file (a private configuration file so we must not put it in the app)

Reply:
ooh ok great thanks